stock_move_actual_date/tests/test_stock_move_actual_date.py

```diff
@@ -0,0 +1,140 @@
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+                "actual_date": actual_date,
+            }
+        )
+
+        move = self.env["stock.move"].create(
+            {
+                "picking_id": receipt.id,
+                "name": "10 in",
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "product_id": self.product_1.id,
+                "product_uom_qty": 10.0,
+                "price_unit": 10,
+                "move_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_1.id,
+                            "location_id": self.supplier_location.id,
+                            "location_dest_id": self.stock_location.id,
+                            "qty_done": 10.0,
+                        },
+                    )
+                ],
+            }
+        )
+        move._action_confirm()
+        move._action_done()
+        return receipt, move
+
+    def create_scrap(self, receipt, actual_date=False):
+        scrap = self.env["stock.scrap"].create(
+            {
+                "product_id": self.product_1.id,
+                "scrap_qty": 2.0,
+                "picking_id": receipt.id,
+                "actual_date": actual_date,
+            }
+        )
+        scrap.action_validate()
+        return scrap
+
+    def test_stock_move_actual_date(self):
+        receipt, move = self.create_picking(date(2024, 9, 1))
+        self.assertEqual(move.actual_date, date(2024, 9, 1))
+        self.assertEqual(
+            move.stock_valuation_layer_ids.account_move_id.date, date(2024, 9, 1)
+        )
+        receipt.actual_date = date(2024, 8, 1)
+        self.assertEqual(move.actual_date, date(2024, 8, 1))
+        self.assertEqual(
+            move.stock_valuation_layer_ids.account_move_id.date, date(2024, 8, 1)
+        )
+        scrap = self.create_scrap(receipt, date(2024, 9, 10))
+        self.assertEqual(scrap.move_id.actual_date, date(2024, 9, 10))
+        self.assertEqual(
+            scrap.move_id.stock_valuation_layer_ids.account_move_id.date,
+            date(2024, 9, 10),
+        )
+        scrap.actual_date = date(2024, 8, 11)
+        self.assertEqual(scrap.move_id.actual_date, date(2024, 8, 11))
+        self.assertEqual(
+            scrap.move_id.stock_valuation_layer_ids.account_move_id.date,
+            date(2024, 8, 11),
+        )
+
+    def test_inventory_adjustment_actual_date(self):
+        _, _ = self.create_picking()
+        inventory_quant = self.env["stock.quant"].search(
+            [
+                ("location_id", "=", self.stock_location.id),
+                ("product_id", "=", self.product_1.id),
+            ]
+        )
+        inventory_quant.inventory_quantity = 20.0
+        inventory_quant.accounting_date = date(2024, 7, 1)
```

Comment:
Why don't we just directly create a quant?

Reply:
There was no reason. Your suggestion is more clear. I updated.